packages/babel-parser/src/types.ts

```diff
@@ -1754,6 +1755,12 @@ export interface TsMappedType extends TsTypeBase {
   nameType: TsType | undefined | null;
 }
 
+export interface TsTemplateLiteralType extends TsTypeBase {
+  type: "TSTemplateLiteralType";
+  quasis: TemplateElement[];
+  expressions: TsType[];
```

Comment:
I overlooked that `TSTemplateLiteralType` uses `types` here. I will update the AST and the generator methods.